src/utils/__tests__/adjust-comments-on-sorted-nodes.spec.ts

```diff
@@ -75,16 +75,16 @@ test('it does not move comments at before all import declarations', () => {
     expect(importNodes).toHaveLength(3);
     const finalNodes = [importNodes[2], importNodes[1], importNodes[0]];
     const adjustedNodes = adjustCommentsOnSortedNodes(importNodes, finalNodes);
-    expect(adjustedNodes).toHaveLength(3);
-    expect(leadingComments(adjustedNodes[0])).toEqual([
-        ' comment c1',
-        ' comment c2',
-    ]);
-    expect(trailingComments(adjustedNodes[0])).toEqual([]);
-    expect(leadingComments(adjustedNodes[1])).toEqual([]);
-    expect(trailingComments(adjustedNodes[1])).toEqual([]);
+    expect(adjustedNodes).toHaveLength(4);
+    // Comment c1 is explicitly detached so it stays with the top-of-file
```

Comment:
Why doesn't comment c1 move?  In your note in the readme, it says"

>If you leave a gap after a comment at the top of your file, we will avoid moving it around if the imports below it shift.

So I would assume since there's no gap, that c1 and c2 would both move with the import from "c".

Reply:
Sorry, I see how I confused things particularly with my variable names and the term “gap”.

I didn’t mean “gap” as in blank line — I have to consider each comment independently (or I need even more richness in my data model/scan-passes). — So gap is “if this comment and the import-node were the only things that existed, would there be a gap?”

So “leading gap” really is “is this specific 1-line comment starting

Reply:
Renamed the variables and comments, in particular because the change from https://github.com/IanVS/prettier-plugin-sort-imports/pull/71#discussion_r1191887394 also further twisted the meaning of one of them.